internal/cmd/validate.go

```diff
@@ -90,70 +90,98 @@ func validatePreRunE(cmd *cobra.Command, _ []string) error {
 }
 
 func validateCmdFunc(cmd *cobra.Command, args []string) error {
-	// Parse the URL of the validation document to import.
-	u, err := url.Parse(args[0])
-	if err != nil {
-		return err
-	}
+	// Initialize variables for multiple files
+	var (
+		onlySchemaFiles            = true
+		validateContentsFiles      [][]byte
+		parsedFiles                []validationfile.ValidationFile
+		totalAssertions            int
+		totalRelationsValidated    int
+		totalFiles                 = len(args)
+		successfullyValidatedFiles = 0
+	)
 
-	decoder, err := decode.DecoderForURL(u)
-	if err != nil {
-		return err
-	}
+	for _, arg := range args {
+		u, err := url.Parse(arg)
+		if err != nil {
+			return err
+		}
 
-	// Decode the validation document.
-	var parsed validationfile.ValidationFile
-	validateContents, isOnlySchema, err := decoder(&parsed)
-	if err != nil {
-		var errWithSource spiceerrors.WithSourceError
-		if errors.As(err, &errWithSource) {
-			ouputErrorWithSource(validateContents, errWithSource)
+		decoder, err := decode.DecoderForURL(u, args)
+		if err != nil {
+			return err
 		}
 
-		return err
+		var parsed validationfile.ValidationFile
+		validateContents, isOnlySchema, err := decoder(&parsed)
+		if err != nil {
+			var errWithSource spiceerrors.WithSourceError
+			if errors.As(err, &errWithSource) {
+				ouputErrorWithSource(validateContents, errWithSource)
+			}
+			return err
+		}
+
+		onlySchemaFiles = isOnlySchema && onlySchemaFiles
+
+		// Store the contents and parsed file
+		validateContentsFiles = append(validateContentsFiles, validateContents)
+		parsedFiles = append(parsedFiles, parsed)
 	}
 
-	// Create the development context.
+	// Create the development context for all parsed files
 	ctx := cmd.Context()
-	tuples := make([]*core.RelationTuple, 0, len(parsed.Relationships.Relationships))
-	for _, rel := range parsed.Relationships.Relationships {
-		tuples = append(tuples, rel.ToCoreTuple())
+	tuples := make([]*core.RelationTuple, 0)
+	for _, parsed := range parsedFiles {
+		for _, rel := range parsed.Relationships.Relationships {
+			tuples = append(tuples, rel.ToCoreTuple())
+		}
 	}
 	devCtx, devErrs, err := development.NewDevContext(ctx, &devinterface.RequestContext{
-		Schema:        parsed.Schema.Schema,
+		Schema:        parsedFiles[0].Schema.Schema,
```

Comment:
I don't that people will use it this way, but choosing the first file and using it for the schema is a hidden behavior - if i had two different schema files in my two different validations, it would error in a confusing way. Is there a way to include this dev context construction as a part of the loop?

Reply:
Like I think it might make sense to put a loop over all of this logic rather than making each block loop individually.